From 7c16ae12ba82aa257003c720fbcabe5d9bb8c464 Mon Sep 17 00:00:00 2001
From: pivaldi <git@ivaldi.me>
Date: Wed, 12 Feb 2025 19:46:01 +0100
Subject: [PATCH] fix(:completion): Remove undefined completion

Fix this error when using "Find file":
Error in post-command-hook (vertico--exhibit):
  (error "Invalid completion style +vertico-basic-remote")
Remove undefined style completion type +vertico-basic-remote
Relative to this comment:
URL_TOO_LONG /issues/7491#issuecomment-2012682178

* modules/completion/vertico/config.el :
  remove `+vertico-basic-remote` use as it's not defined.

Fix: #7491
---
 modules/completion/vertico/config.el | 19 +++++++++----------
 1 file changed, 9 insertions(+), 10 deletions(-)

diff --git a/modules/completion/vertico/config.el b/modules/completion/vertico/config.el
index ee98130c84e..159eb3639ba 100644
--- a/modules/completion/vertico/config.el
+++ b/modules/completion/vertico/config.el
@@ -88,15 +88,14 @@ orderless."
 
   (add-to-list
    'completion-styles-alist
-   '(+vertico-basic-remote
-     +vertico-basic-remote-try-completion
+   '(+vertico-basic-remote-try-completion
      +vertico-basic-remote-all-completions
      "Use basic completion on remote files only"))
   (setq completion-styles '(orderless basic)
         completion-category-defaults nil
         ;; note that despite override in the name orderless can still be used in
         ;; find-file etc.
-        completion-category-overrides '((file (styles +vertico-basic-remote orderless partial-completion)))
+        completion-category-overrides '((file (styles orderless partial-completion)))
         orderless-component-separator #'orderless-escapable-split-on-space)
   ;; ...otherwise find-file gets different highlighting than other commands
   (set-face-attribute 'completions-first-difference nil :inherit nil))
@@ -377,13 +376,13 @@ orderless."
     "If MODE is enabled, highlight it as font-lock-constant-face."
     (let ((sym (intern cmd)))
       (with-current-buffer (nth 1 (buffer-list))
-      (if (or (eq sym major-mode)
-              (and
-               (memq sym minor-mode-list)
-               (boundp sym)
-               (symbol-value sym)))
-          (add-face-text-property 0 (length cmd) 'font-lock-constant-face 'append cmd)))
-        cmd))
+        (if (or (eq sym major-mode)
+                (and
+                 (memq sym minor-mode-list)
+                 (boundp sym)
+                 (symbol-value sym)))
+            (add-face-text-property 0 (length cmd) 'font-lock-constant-face 'append cmd)))
+      cmd))
 
   (add-to-list 'vertico-multiform-categories
                '(file
